crossbeam-channel/src/channel.rs

```diff
@@ -656,14 +656,52 @@ impl<T> Sender<T> {
             _ => false,
         }
     }
+
+    /// Disconnects the channel.
+    ///
+    /// Explicitly disconnects the channel and returns `true`, as if all instances of either side
+    /// (sender or receiver) are dropped, unless it has been already disconnected. Otherwise, this
+    /// method does nothing and returns `false`.
+    ///
+    /// The successful disconnect operation results in immediately rejecting sending subsequent
+    /// messages to receivers.
+    ///
+    /// Disconnected channels can be restored connected by calling [`connect`], as long as there
+    /// are at least a sender and a receiver for the channel.
+    ///
```

Comment:
i guess i need to add some motivating example here for the justification of disconnect()...